bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/WalletRpc.scala

```diff
@@ -264,14 +419,60 @@ trait WalletRpc { self: Client =>
           List(JsString(walletName), JsBoolean(disablePrivateKeys)))
     }
 
-  def getAddressInfo(address: BitcoinAddress): Future[AddressInfoResult] = {
+  def getAddressInfo(
+      address: BitcoinAddress,
+      walletNameOpt: Option[String] = None): Future[AddressInfoResult] = {
     self.version match {
       case V16 | V17 =>
-        bitcoindCall[AddressInfoResultPreV18]("getaddressinfo",
-                                              List(JsString(address.value)))
+        bitcoindCall[AddressInfoResultPreV18](
+          "getaddressinfo",
+          List(JsString(address.value)),
+          uriExtensionOpt = walletNameOpt.map(walletExtension))
       case V18 | V19 | V20 | Experimental | Unknown =>
-        bitcoindCall[AddressInfoResultPostV18]("getaddressinfo",
-                                               List(JsString(address.value)))
+        bitcoindCall[AddressInfoResultPostV18](
+          "getaddressinfo",
+          List(JsString(address.value)),
+          uriExtensionOpt = walletNameOpt.map(walletExtension))
     }
   }
+
+  def sendMany(
+      amounts: Map[BitcoinAddress, CurrencyUnit],
+      minconf: Int = 1,
+      comment: String = "",
+      subtractFeeFrom: Vector[BitcoinAddress] = Vector.empty,
+      walletNameOpt: Option[String] = None): Future[DoubleSha256DigestBE] = {
+    val jsonOutputs: JsValue = Json.toJson {
+      amounts.map {
+        case (addr, curr) => addr -> Bitcoins(curr.satoshis)
+      }
+    }
+    bitcoindCall[DoubleSha256DigestBE](
+      "sendmany",
+      List(JsString(""),
+           jsonOutputs,
+           JsNumber(minconf),
+           JsString(comment),
+           Json.toJson(subtractFeeFrom)),
+      uriExtensionOpt = walletNameOpt.map(walletExtension)
+    )
+  }
+
+  def sendToAddress(
+      address: BitcoinAddress,
+      amount: CurrencyUnit,
+      localComment: String = "",
+      toComment: String = "",
+      subractFeeFromAmount: Boolean = false,
+      walletNameOpt: Option[String] = None): Future[DoubleSha256DigestBE] = {
+    bitcoindCall[DoubleSha256DigestBE](
+      "sendtoaddress",
+      List(Json.toJson(address),
+           Json.toJson(Bitcoins(amount.satoshis)),
+           JsString(localComment),
+           JsString(toComment),
+           JsBoolean(subractFeeFromAmount)),
+      uriExtensionOpt = walletNameOpt.map(walletExtension)
+    )
+  }
```

Comment:
So just looking through this file, if we were to move the wallet name into `BitcoindRpcAppConfig` could we avoid all of the duplication of methods? 

Basically all of these would now just look inside of `BitcoindRpcAppConfig` and get the wallet name if it's defined.

Another reason this is probably a good idea is that authentication and having access to a wallet is inherently tied together, so with the `username`/`password` combo in `BitcoindRpcAppConfig` that should authenticate you with a specific wallet which should also be defined in `BitciondRpcAppConfig`?

Reply:
Bitcoind's rpc auth isn't tied to each wallet so I don't see the benefit there.